src/transformStateWithClones.js

```diff
@@ -7,7 +7,60 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const tmpState = {};
+
+  Object.assign(tmpState, state);
+
+  const arrRes = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        addProperties(tmpState, action.extraData);
+
+        const obj1 = {};
+
+        assignObj(obj1, tmpState, arrRes);
+        break;
```

Comment:
don't need to write a lot of extra logic, it can be simpler 
check other cases
```suggestion
      case 'addProperties':
        Object.assign(tmpState, extraData);
        break;
```